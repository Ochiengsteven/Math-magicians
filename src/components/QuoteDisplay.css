/* QuoteDisplay.css */

.quote-display {
  padding: 20px;
  border: 1px solid #ddd;
  background-color: #f5f5f5;
  text-align: center;
  margin-top: 20px;
  max-width: 400px;
  margin-left: auto;
  margin-right: auto;
}

.quote-title {
  font-size: 1.5rem;
  margin-bottom: 10px;
}

.quote-text {
  font-size: 1.2rem;
  font-style: italic;
}

.author-text {
  font-size: 1rem;
  margin-top: 10px;
  text-align: right;
}
