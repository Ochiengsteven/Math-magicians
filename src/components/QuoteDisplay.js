import React, { useState, useEffect } from 'react';
import './QuoteDisplay.css'; // Import your CSS file for styling

const QuoteDisplay = () => {
  const [quote, setQuote] = useState(null);
  const [author, setAuthor] = useState(null); // Define author state
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);
  const category = 'men';
  const apiKey = 'dZB1jc2cSzXdURZ2PHCOdA==2ejjLjTFQHM1u8lb';

  useEffect(() => {
    fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
      headers: { 'X-Api-Key': apiKey },
    })
      .then((response) => response.json())
      .then((data) => {
        const firstQuote = data[0];
        setQuote(firstQuote?.quote);
        setAuthor(firstQuote?.author); // Set the author from API response
        setIsLoading(false);
      })
      .catch((error) => {
        setError(error);
        setIsLoading(false);
      });
  }, []);

  if (isLoading) {
    return <div className="quote-display loading">Loading...</div>;
  }

  if (error) {
    return (
      <div className="quote-display error">
        Error:
        {' '}
        {error.message}
      </div>
    );
  }

  return (
    <div className="quote-display">
      <p className="quote-text">
        &quot;
        {quote}
        &quot;
      </p>
      <p className="author-text">
        -
        {author}
      </p>
    </div>
  );
};

export default QuoteDisplay;
