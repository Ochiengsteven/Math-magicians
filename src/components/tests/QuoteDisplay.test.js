import React from 'react';
import { render } from '@testing-library/react';
import QuoteDisplay from '../QuoteDisplay';

describe('QuoteDisplay component', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<QuoteDisplay />);
    expect(asFragment()).toMatchSnapshot();
  });
});
