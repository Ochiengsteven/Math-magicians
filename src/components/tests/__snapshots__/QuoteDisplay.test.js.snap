// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteDisplay component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="quote-display loading"
  >
    Loading...
  </div>
</DocumentFragment>
`;
